Jantimizer/QueryTestSuite/Parsers/BasePlanParser.cs

```diff
@@ -10,7 +10,8 @@ namespace QueryTestSuite.Parsers
 {
     public abstract class BasePlanParser : IPlanParser
     {
-        public abstract AnalysisResult ParsePlan(DataSet planData);
+        public abstract Task<AnalysisResult> ParsePlan(DataSet planData);
+        public abstract Task<AnalysisResult> ParsePlan(Task<DataSet> planData);
```

Comment:
Why make a task result, that takes a task as input?
Would it not make more sense to await earlier in the code instead?